### DISCUSSION TITLE
Add an option for preventing automatic user creation on OAuth sign-in

### DISCUSSION THREAD

**Discussion by tiguchi** [gravitl/netmaker#1367] [Ideas]
Hi! I just set up a Google OAuth provider in Netmaker and noticed that there is no setting to prevent automatic user account creation on sign-in via OAuth provider.

I understand that a user has zero privileges in that case, until approved by an admin, but I'd rather have a setting that would only allow already existing user accounts to utilize OAuth login, or a setting for switching off new account creation after authorized users signed in via OAuth provider.

**Comment by afeiszli** [comment]
You would need to either configure your OAuth with Google to handle particular users, or use an intermediary with generic OIDC, like Auth0.
